src/script_opt/ProfileFunc.cc

```diff
@@ -859,6 +863,12 @@ void ProfileFuncs::ComputeProfileHash(std::shared_ptr<ProfileFunc> pf) {
     // prevent collisions due to elements with simple hashes
     // (such as Stmt's or Expr's that are only represented by
     // the hash of their tag).
+    h = merge_p_hashes(h, p_hash("params"));
```

Comment:
Do you mean why is all that other stuff also hashed? If so, it's because for `-O gen-C++` there's a need to avoid the fairly baffling situation where a script is compiled, then altered, and `-O use-C++` picks up the old code rather than the new. So the goal is to produce a comprehensive hash over every aspect of the function. In the situation where the script is altered, the new code will be picked up (and interpreted). The user can tell whether that's happening by using `-O report-C++`.

Reply:
That helps, thanks, though I meant something more mundane — changes like
```
if ( compute_func_hashes || ! pf->HasHashVal() )
            ComputeProfileHash(f.ProfilePtr());
```
seem to be there to ensure the hash isn't already there. (Minor one way or another, so let me resolve this one.)